routes/api/farms.php

```diff
@@ -4,81 +4,102 @@
 // > GET /api/farms/{id}
 // ==================================================
 $app->get('/api/farms/{id}', function ($request, $response, $args) {
-  $farm = FarmQuery::create()->findPK($args['id']);
-  if ($farm) {
-    return $response->withJson($farm->toArray(), 200);
-  } else {
-    return $response->withStatus(404);
-  }
+  return api\view($response, FarmQuery::create()->findPK($args['id']));
 });
 
 // ==================================================
 // > GET /api/farms/
 // ==================================================
 $app->get('/api/farms/', function ($request, $response) {
-  $farms = FarmQuery::create()->find()->toArray();
-  return $response->withJson($farms, 200);
+  $farms = FarmQuery::create()->find();
+  return api\mapCollection(
+    $response, $farms,
+    function ($farm) use ($request) {
+      $data = $farm->serialize();
+      if (auth\isUser($request)) {
+        $user = auth\getUser($request);
+        $data["subscribed"] = $farm->hasSubscriber($user);
+      }
+      return $data;
+    }
+  );
 });
 
 // ==================================================
 // > POST /api/farms/
 // ==================================================
 $app->post('/api/farms/', function ($request, $response) {
```

Comment:
I don't mind anyone being able to create a farm now, but this is something we need to correct in the future.